src/include/mpi.h.in

```diff
@@ -806,6 +811,11 @@ typedef int (MPI_Delete_function) ( MPI_Comm, int, void *, void * );
 /* User combination function */
 typedef void (MPI_User_function) ( void *, void *, int *, MPI_Datatype * );
 typedef void (MPI_User_function_c) ( void *, void *, MPI_Count *, MPI_Datatype * );
+/* the _x callbacks accepts scalar inputs and an additional "void *" extra_state */
+typedef void (MPIX_User_function_x) ( void *, void *, MPI_Count, MPI_Datatype, void *);
+
+/* User destructor */
+typedef void (MPIX_Destructor_function) (void *);
```

Comment:
MPI does not use the word "destructor" much in its parlance. Maybe we need something more commonly used, like "MPIX_Free_function" or "MPIX_User_free_function"? We have a precedent for such naming convention: `MPI_Grequest_free_function`.

Reply:
I am open on this.